srcdeps-core/src/main/java/org/srcdeps/core/config/MavenFailWith.java

```diff
@@ -0,0 +1,236 @@
+/**
+ * Copyright 2015-2016 Maven Source Dependencies
+ * Plugin contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.srcdeps.core.config;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.Stack;
+
+import org.srcdeps.core.config.tree.ListOfScalarsNode;
+import org.srcdeps.core.config.tree.Node;
+import org.srcdeps.core.config.tree.ScalarNode;
+import org.srcdeps.core.config.tree.impl.DefaultContainerNode;
+import org.srcdeps.core.config.tree.impl.DefaultListOfScalarsNode;
+import org.srcdeps.core.config.tree.impl.DefaultScalarNode;
+
+/**
+ * A configuration node to define conditions under which the outer Maven build that has source dependencies should fail.
+ * Note that these settings are effective only if source dependencies exist. In case there are no source dependencies in
+ * the outer project, the build will not fail even if any of the goals, profiles or properties are present. This is esp.
+ * handy if you want to avoid releases with source dependencies.
+ *
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+public class MavenFailWith {
+
+    public static class Builder extends DefaultContainerNode<Node> {
+
+        ScalarNode<Boolean> addDefaults = new DefaultScalarNode<>("addDefaults", Boolean.TRUE);
+        ListOfScalarsNode<String> goals = new DefaultListOfScalarsNode<String>("goals", String.class) {
+
+            @Override
+            public void applyDefaultsAndInheritance(Stack<Node> configurationStack) {
+                Builder.this.addDefaults.applyDefaultsAndInheritance(configurationStack);
+                if (Builder.this.addDefaults.getValue() && getElements().isEmpty()) {
+                    addAll(DEFAULT_FAIL_GOALS);
+                }
+            }
+
+            @Override
+            public boolean isInDefaultState(Stack<Node> configurationStack) {
+                return getElements().isEmpty() || (Boolean.TRUE
```

Comment:
Good catch! - the matching right bracket of `equals(` should actually be just after `Builder.this.addDefaults.getValue()` rather than at the very end. That's because `Builder.this.addDefaults.getValue()` can be `null`. Let me fix it.